Dev/adrv9002 unit test

nunojsa opened the pull request:
# Description

This PR adds new tests/functionality to adrv9002 iio driver. It also proposes some fixes on the common conftest which are explained in the commit messages. 

- [ * ] Bug fix (non-breaking change which fixes an issue)
- [ * ] New feature (non-breaking change which adds functionality)

The tests were done on a real adrv9002 platform and confirmed to pass...

* Hardware: adrv9002 evaluation board
* OS: linux

tfcollins commented:
@nunojsa Did you want me to add a profiles directory?

-Travis

nunojsa commented:
> @nunojsa Did you want me to add a profiles directory?
> 
> -Travis

:facepalm: I did create the directory with a couple of profiles. Just forgot to commit them. Will wait a bit for some other things that I might need to change and include it in v2 of the PR

tfcollins commented:
Can you also run the pre-commit checks (invoke precommit). I think mypy will complain about the profile property definition.

tfcollins commented on test/conftest.py:
```diff
@@ -262,6 +266,43 @@ def attribute_single_value_pow2(classname, devicename, attr, max_pow, tol, repea
+                assert bi.dev_interface(val, attr, tol) <= tol
+
+
+def attribute_profile_write(classname, devicename, attr, file):
```
Can we make this a little more generic? Maybe call it attribute_write_only_str or something.

tfcollins commented on adi/adrv9002.py:
```diff
@@ -113,17 +113,14 @@ def __init__(self, uri=""):
-
-    @profile.setter
     def profile(self, value):
         with open(value, "r") as file:
```
Need a docstring

nunojsa commented:
Changes in v2:

- Run `pre-commit` and fix the reported failure;

- Added docstring to  `write_profile`;

- Renamed `attribute_profile_write` to `attribute_write_only_str`;

- Fixed adrv9002_examples.py since thereS is no rx_sample_rate attribute on the adrv9002 class;

- Added adrv9002 profiles folder.

tfcollins commented:
Look great! Thanks

-Travis
